Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -159,9 +199,14 @@ public struct MongoConnectionString: Codable, LosslessStringConvertible {
         fileprivate var tlsDisableCertificateRevocationCheck: Bool?
         fileprivate var tlsDisableOCSPEndpointCheck: Bool?
         fileprivate var tlsInsecure: Bool?
+        fileprivate var w: WriteConcern.W?
+        fileprivate var wTimeoutMS: Int?
+        fileprivate var zlibCompressionLevel: Int32?
 
         fileprivate init(_ uriOptions: String) throws {
             let options = uriOptions.components(separatedBy: "&")
+            // tracks options that have already been set to error on duplicates
+            var setOptions: [String] = []
```

Comment:
rather than using an array here, we can use a [`Set`](https://developer.apple.com/documentation/swift/set) since we'll be doing mostly inclusion lookups.